### ISSUE TITLE
iterate FalcorJSON instance

### ISSUE THREAD

**Issue by jameslaneconkling** [graphistry/falcor#10]
(continues discussion at end of issue thread #8)

So, the use case is iterating a `FalcorJSON` response instance while avoiding metadata.  As @trxcllnt points out, b/c the `FalcorJSON` class [copies methods from the `Array.prototype`](https://github.com/graphistry/falcor/blob/master/packages/falcor/lib/cache/get/json/FalcorJSON.js#L51), you can use map/reduce/filter/etc. assuming the `FalcorJSON` instance has a numeric length property.

However, this doesn't work if using [`boxValues`](http://netflix.github.io/falcor/doc/Model.html#boxValues) to make requests, as length resolves to an Object (e.g. `{ $size: 51, $type: "atom", value: 1221 }`).

I was considering overwriting the `FalcorJSON` methods to use the length boxedValue at `length.value`, using [a polyfill as a guide ](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map), but realized that for cases where I was working with graph fragments with large indices, (e.g. `['todos', { from: 1000, to: 1010 }, ...]`), there would be unnecessary overhead.

So, instead I wrote a pretty naive utility function

```javascript
const mapFalcorJSON2Array = (project, from, to, falcorJSON) => {
  const result = [];

  while (from <= to) {
    result.push(project(falcorJSON[from], from));
    from += 1;
  }

  return result;
};

// e.g.
mapFalcorJSON2Array((todo, idx) => ({ ...todo, idx }), 1000, 1010, todosFalcorJSON)
```

Not sure if there are better/more general solutions.  @trxcllnt thoughts.

**Comment by trxcllnt** [comment]
@jameslaneconkling That looks good to me. I initially thought about adding a method like this to the prototype, but realized `slice` also does the job: `falcorList.slice(1000, 1010).map(() => )`.

As for boxed values, the only solution I saw was calling each Array method with the boxed length value, essentially changing [this function](https://github.com/graphistry/falcor/blob/master/packages/falcor/lib/cache/get/json/FalcorJSON.js#L67) to read:

```javascript
function bindArrayMethod(name, fn) {
    return new Function('fn', 'aSlice',
        'return function ' + name + ' () {' +
            'var length = this.length;' +
            '(typeof length === "object") && (length = length.value);' +
            'return typeof length !== "number" ? [] :' +
                'fn.apply(aSlice.call(this, 0, length), arguments);' +
        '};'
    )(fn, Array.prototype.slice)
}
```

**Comment by trxcllnt** [comment]
@jameslaneconkling Just published `2.7.0` with [this fix](https://github.com/graphistry/falcor/blob/22d525600bd5f9436fe13d6941c10fb7d5682f99/packages/falcor/lib/cache/get/json/FalcorJSON.js#L67)

**Comment by jameslaneconkling** [comment]
Oh, this is very cool--thanks for the update.  Just took it for a test and it works as expected when length is an atom.

I think that closes this out.  Feel free to reopen if you have any additional thoughts/updates on the matter.
